MovieNight/MovieMatcher.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  MovieMatcher.swift
+//  MovieNight
+//
+//  Created by Tassia Serrao on 15/07/2017.
+//  Copyright © 2017 Tassia Serrao. All rights reserved.
+//
+
+import Foundation
+
+protocol Matcher {
+    func matchUserSelections(with foxGenres: [Genre], and crabGenres: [Genre], and foxActors: [Actor], and crabActors: [Actor], and crabMovies: [Movie], and foxMovies: [Movie], completionHandler: @escaping ([Movie]) -> Void)
+}
+
+class MovieNightMatcher: Matcher {
+    
+    var apiCLient = Factory.createApiClient()
+    var recommendedMovies = [[Movie]]()
+    var moviesMatchedByGenre: [Movie] = [Movie]()
+    var moviesMatchedByActor: [Movie] = [Movie]()
+    var allRecommendedMovies = [Movie]()
+    var moviesCredits = [[Credit]]()
+    var matchedMoviesByGenreAndActor: [Movie] = [Movie]()
+
+    func matchUserSelections(with foxGenres: [Genre], and crabGenres: [Genre], and foxActors: [Actor], and crabActors: [Actor], and crabMovies: [Movie], and foxMovies: [Movie], completionHandler: @escaping ([Movie]) -> Void) {
+        
+        let moviesSelected = foxMovies + crabMovies
+        
+        // First: get movies recommendations from user's movies selections
+        getMovieRecommendations(from: moviesSelected, completionHandler: { movies in
+            
+            //convert to [Movies] instead of [[Movie]]
+            self.allRecommendedMovies = Array(movies.joined())
```

Comment:
It would be nicer if `movies` was already a flat array. Then this step wouldn't be needed. (see comment below)